src/scores/categorical/risk_matrix_impl.py

```diff
@@ -0,0 +1,498 @@
+    prob_threshold_dim: str,
+    prob_threshold_coords: Iterable,
+) -> xr.DataArray:
+    """
+    Given a warning scaling matrix, evaluation weights and other inputs,
+    returns the weights for each risk matrix decision point as an xarray data array. The returned
+    data array is designed to be used for the :py:func:`scores.categorical.risk_matrix_score` calculation.
+
+    Comprehensive checks are made on ``scaling_matrix`` to ensure it satisfies the properties
+    of warning scaling in Table 1 of Taggart & Wilke (2025).
+
+    Args:
+        scaling_matrix: a 2-dimensional matrix encoding the warning scaling. Warning levels
+            are given integer values 0, 1, ..., q. The top row corresponds to the
+            highest certainty category while the right-most column corresponds to most
+            severe category.
+        evaluation_weights: positive weights used for warning evaluation. The kth weight
+            (corresponding to ``evaluation_weights[k-1]``) is proportional to the importance
+            of accurately discriminating between warning states below level k and
+            warning states at or above level k. ``len(evaluation_weights)`` must be at least q.
+        severity_dim: name of the severity category dimension.
+        severity_coords: labels for each of the severity categories, in order of increasing
+            severity. Does NOT include the lowest severity category for which no warning would
+            be issued.
+        prob_threshold_dim: name of the probability threshold dimension.
+        prob_threshold_coords: list of the probability decision thresholds in the risk matrix,
+            strictly between 0 and 1.
+
+    Returns:
+        xarray data array of weights for each risk matrix decision point, indexed by prob_threshold_dim
+        and severity_dim.
+
+    Raises:
+        ValueError: if ``matrix_weights`` isn't two dimensional.
+        ValueError: if ``scaling_matrix`` has entries that are not non-negative integers.
+        ValueError: if the first column or last row of ``scaling_matrix`` has nonzero entries.
+        ValueError: if ``scaling_matrix`` decreases along any row (moving left to right).
+        ValueError: if ``scaling_matrix`` increases along any column (moving top to bottom).
+        ValueError: if number of rows of ``matrix_weights`` doesn't equal length of ``prob_threshold_coords``.
+        ValueError: if number of columns of ``matrix_weights`` doesn't equal length of ``severity_coords``.
+        ValueError: ``len(evaluation_weights)`` is less than the maximum value in ``scaling_matrix``.
+        ValueError: if ``prob_threshold_coords`` aren't strictly between 0 and 1.
+        ValueError: if ``evaluation_weights`` aren't strictly positive.
+
+    References:
+        - Taggart, R. J., & Wilke, D. J. (2025). Warnings based on risk matrices: a coherent framework
+          with consistent evaluation. Nat. Hazards Earth Syst. Sci., 25, 2657–2677.
```

Comment:
```suggestion
          with consistent evaluation. Natural Hazards and Earth System Sciences, 25(8), 2657–2677.
```
As above - suggest adding full journal title & issue number